src/Kaleidoscope-Simulator.h

```diff
@@ -0,0 +1,70 @@
+/* -*- mode: c++ -*-
```

Comment:
Like in the case of `src/Aglais.h`, I'd move this somewhere else too. `src/kaleidoscope/simulator/` sounds like a good place, to match the namespace.